Services/Certificate/classes/API/Download/class.ilUserCertificateApiGUI.php

```diff
@@ -46,10 +52,34 @@ public function __construct(
         }
         $this->certificateLogger = $certificateLogger;
 
+        if ($controller === null) {
+            $controller = $DIC->ctrl();
+        }
+        $this->controller = $controller;
+
 
         $this->language->loadLanguageModule('cert');
     }
 
+
+    /**
+     *
+     */
+    public function executeCommand() : void
```

Comment:
Thanks a lot. This was indeed missing. Niels tested this by calling `download` directly in the consumer, so the absence was not really obvious.